### PR TITLE
Profiling w/ `tracing`

### PR DESCRIPTION
**Pull request by kirawi**
- [ ] `helix-lsp`
- [ ] `helix-view`
- [ ] `helix-term`
- [ ] Default `log` subscriber & remove `fern`
- [ ] Support other feature-gated subscribers

### CODE REVIEW

File: helix-term/Cargo.toml [hunk 1]
```diff
@@ -25,6 +25,11 @@ helix-core = { version = "0.3", path = "../helix-core" }
 helix-view = { version = "0.3", path = "../helix-view" }
 helix-lsp = { version = "0.3", path = "../helix-lsp" }
 
+tracing = "0.1"
+tracing-subscriber = "0.2"
+tracing-tracy = "0.6"
+tracing-flame = "0.1"
```
Comment (archseer) [inline helix-term/Cargo.toml:31]: https://docs.rs/tracing-tracy/0.6.1/tracing_tracy/#important-note

Both of these should be marked as optional and placed under separate features that aren't enabled by default.

File: helix-lsp/src/client.rs [hunk 2]
```diff
@@ -28,6 +28,7 @@ pub struct Client {
 }
 
 impl Client {
+    #[tracing::instrument(name = "Client::start")]
```
Comment (archseer) [inline helix-lsp/src/client.rs:31]: I don't know if the name is necessary on any of these, tracing will use the method name by default. The module will be used as the target.

> By default, the generated span’s name will be the name of the function, the span’s target will be the current module path, and the span’s level will be INFO

**Review by pickfire — COMMENTED** [review]
Shouldn't the log stuff be replaced by tracing as well?

**Comment by kirawi** [comment]
> Shouldn't the log stuff be replaced by tracing as well?

Correct me if I'm wrong, but I thought that `tracing` is for collecting performance data and such? I know it can also collect ["events"](https://tracing.rs/tracing/#events), but they don't seem to right for this purpose, at least to me? Not sure. I'll also make `tracing-log` the default, since it's included with `tracing-subscriber` by default, so I don't know if I'll have to do that anyway.

**Comment by oberblastmeister** [comment]
@kirawi Are you still working on this? If you are not I'd like to take this on if you are okay with that.

**Comment by kirawi** [comment]
> @kirawi Are you still working on this? If you are not I'd like to take this on if you are okay with that.

Yeah sure, go ahead! When I was still working on this, I was looking into replacing `fern` with `tracing` for logging as well.

**Pull request closed without merging**